src/key_packages/mod.rs

```diff
@@ -237,6 +237,22 @@ impl KeyPackageBundle {
         Self::new_from_leaf_secret(ciphersuites, credential_bundle, extensions, leaf_secret)
     }
 
+    /// Replace the init key in the `KeyPackage` with a random one and return a
+    /// `KeyPackageBundle` with the corresponding secret values
+    pub fn from_rekeyed_key_package(key_package: &KeyPackage) -> Self {
```

Comment:
Why are we not doing this from a `KeyPackageBundle`? It seems like that's what happened before, but now it's done based on a `KeyPackage`. It seems more natural to me to begin with the private part (the bundle) and then generate the public part (the package) based on that.

Reply:
That's what the function does internally. As @franziskuskiefer mentioned, we could have that function as part of `KeyPackage` instead.